### PR TITLE
feat: [IOBP-9] Add IDPay transaction QRCode scan screen

### PR DESCRIPTION
**Pull request by mastro993**
## Short description
This PR adds the screen for the IDPay transaction QRCode scan.

## List of changes proposed in this pull request
- ts/features/idpay/payment/components/BarcodeScanner.tsx: 
   - Added `BarcodeScanner` component, whihc allows the scan of a barcode/QR Code
   - Added `useIOBarcodeScanner`, hook that allows to use the `BarcodeScanner` component with a configuration
- ts/features/idpay/payment/screens/IDPayPaymentCodeScanScreen.tsx:
   - Added `BarcodeScanner` to the view
   - Added permission views, which shows to the user the right prompt to grant camera permissions
 - Added required locale keys
 - Added required pictograms

## How to test
- Tests should solve correctly.
- With the `io-dev-server`, from the wallet section of the app navigate to details screen of a discount type initiative (marked with a **[D]** in its name). From here, click on the payment authorization button and scan [this QRCode](https://continua.io.pagopa.it/qrcode.png?feat=idpay&trxcode=00000000).

### CODE REVIEW

File: locales/en/index.yml [hunk 1]
```diff
@@ -3650,6 +3650,20 @@ idpay:
         subtitle: Enter the 8 characters shown below the QR code.
         input: Codice Transazione
         button: Continua
+      scan:
+        tabs:
+          scan: Inquadra
+          upload: Carica
```
Comment (thisisjp) [inline locales/en/index.yml:3656]:
```suggestion
          upload: Upload
```

File: locales/en/index.yml [hunk 2]
```diff
@@ -3650,6 +3650,20 @@ idpay:
...
+          input: Digita
```
Comment (thisisjp) [inline locales/en/index.yml:3657]:
```suggestion
          input: Type
```

File: locales/en/index.yml [hunk 3]
```diff
@@ -3650,6 +3650,20 @@ idpay:
...
+        permissions:
+          undefined: 
+            title: Ci serve il tuo permesso
```
Comment (thisisjp) [inline locales/en/index.yml:3660]:
```suggestion
            title: We need your consent
```

File: locales/en/index.yml [hunk 4]
```diff
@@ -3650,6 +3650,20 @@ idpay:
...
+            label: Consenti l’accesso per inquadrare codici QR con la fotocamera del tuo dispositivo.
+            action: Consenti l’accesso
```
Comment (thisisjp) [inline locales/en/index.yml:3662]:
```suggestion
            action: Allow access
```

File: locales/en/index.yml [hunk 5]
```diff
@@ -3650,6 +3650,20 @@ idpay:
...
+          denied:
+            title: Non hai dato il permesso di usare la fotocamera
+            label: Per concederlo, accedi alle impostazioni del tuo sistema operativo.
+            action: Apri impostazioni
```
Comment (thisisjp) [inline locales/en/index.yml:3666]:
```suggestion
            action: Go to settings
```

File: locales/it/index.yml [hunk 6]
```diff
@@ -3673,6 +3673,20 @@ idpay:
...
+          input: Digita
+        permissions:
+          undefined: 
+            title: Ci serve il tuo permesso
+            label: Consenti l’accesso per inquadrare codici QR con la fotocamera del tuo dispositivo.
+            action: Consenti l’accesso
+          denied:
+            title: Non hai dato il permesso di usare la fotocamera
+            label: Per concederlo, accedi alle impostazioni del tuo sistema operativo.
+            action: Apri impostazioni
```
Comment (thisisjp) [inline locales/it/index.yml:3689]:
```suggestion
            action: Vai alle impostazioni
```

File: locales/en/index.yml [hunk 7]
```diff
@@ -3650,6 +3650,20 @@ idpay:
...
         button: Continua
+      scan:
+        tabs:
+          scan: Inquadra
+          upload: Carica
+          input: Digita
+        permissions:
+          undefined: 
+            title: Ci serve il tuo permesso
+            label: Consenti l’accesso per inquadrare codici QR con la fotocamera del tuo dispositivo.
```
Comment (thisisjp) [inline locales/en/index.yml:3661]:
```suggestion
            label: Allow access to scan QR codes with your device camera.
```

File: locales/en/index.yml [hunk 8]
```diff
@@ -3650,6 +3650,20 @@ idpay:
...
+            action: Consenti l’accesso
+          denied:
+            title: Non hai dato il permesso di usare la fotocamera
```
Comment (thisisjp) [inline locales/en/index.yml:3664]:
```suggestion
            title: You didn't give access to camera
```

File: locales/en/index.yml [hunk 9]
```diff
@@ -3650,6 +3650,20 @@ idpay:
...
+            label: Per concederlo, accedi alle impostazioni del tuo sistema operativo.
```
Comment (thisisjp) [inline locales/en/index.yml:3665]:
```suggestion
            label: Browse your device settings to allow camera access.
```

**Review by thisisjp — COMMENTED** [review]
Eng translation (see upcoming commit)

**Review by forrest57 — APPROVED** [review]

**Pull request merged**